src/lib/gossip_net/gossip_net.ml

```diff
@@ -565,6 +541,39 @@ module Make (Message : Message_intf) : S with type msg := Message.msg = struct
               failwithf "Unhandled Membership.connect exception: %s"
                 (Exn.to_string e) ()
         in
+        let peer_event_reader, peer_event_writer = Linear_pipe.create () in
+        Linear_pipe.iter_unordered peer_event_reader ~max_concurrency:64
+          ~f:(fun event ->
+            let%bind t = t_for_restarting in
```

Comment:
this doesn't actually support restarting now right?

Reply:
Restarting should continue to work the same as it has with this PR. The libp2p stuff doesn't support being restarted yet, but the old Haskell stuff does.